src/runtime/statistics.py

```diff
@@ -62,3 +133,187 @@ def harmonic_mean(x: list[i32]) -> f64:
     ans: f64
     ans = k/sum
     return ans
+
+@overload
+def median(x: list[i32]) -> f64:
+    """
+    Returns the median of a data sequence of numbers
+    """
+    k: i32 = len(x)
+    if k == 0:
+        return 0.0
+    i: i32
+    j: i32
+    for i in range(k):
+        for j in range(i+1,k):
+            if x[i]>x[j]:
+                x[i],x[j]=x[j],x[i]
+    ans: f64
+    if k%2==1:
+        medianidx: i32
+        medianidx = k/2
+        ans = float(x[medianidx])
+    else:
+        lowmedidx: i32
+        highmedidx: i32
+        lowmedidx = (k/2)-1
+        highmedidx = k/2
+        ans = float((x[lowmedidx] + x[highmedidx])/2)
+    return ans
+
+@overload
+def median(x: list[f64]) -> f64:
+    """
+    Returns the median of a data sequence of numbers
+    """
+    k: i32 = len(x)
+    if k == 0:
+        return 0.0
+    i: i32
+    j: i32
+    for i in range(k):
+        for j in range(i+1,k):
+            if x[i]>x[j]:
+                x[i],x[j]=x[j],x[i]
+    ans: f64
+    if k%2==1:
+        medianidx: i32
+        medianidx = k/2
+        ans = x[medianidx]
+    else:
+        lowmedidx: i32
+        highmedidx: i32
+        lowmedidx = (k/2)-1
+        highmedidx = k/2
+        ans = (x[lowmedidx] + x[highmedidx])/2
+    return ans
+
+@overload
+def median(x: list[i32]) -> i32:
+    """
+    Returns the median of a data sequence of numbers
+    """
+    k: i32 = len(x)
+    if k == 0:
+        return 0
+    i: i32
+    j: i32
+    for i in range(k):
+        for j in range(i+1,k):
+            if x[i]>x[j]:
+                x[i],x[j]=x[j],x[i]
+    ans: i32
+    if k%2==1:
+        medianidx: i32
+        medianidx = k/2
+        ans = x[medianidx]
+        return ans
+    else:
+        lowmedidx: i32
+        highmedidx: i32
+        lowmedidx = (k/2)-1
+        highmedidx = k/2
+        if (x[lowmedidx] + x[highmedidx])%2 ==0:
+            ans = (x[lowmedidx] + x[highmedidx])/2
+            return ans
+
+@overload
+def median_low(x: list[i32]) -> i32:
+    """
+    Returns the lowmedian of a data sequence of numbers
+    """
+    k: i32 = len(x)
+    if k == 0:
+        return 0
+    i: i32
+    j: i32
+    for i in range(k):
+        for j in range(i+1,k):
+            if x[i]>x[j]:
+                x[i],x[j]=x[j],x[i]
```

Comment:
This code is repeated many times. How about creating a new function, say `_sort` and reuse it?

Reply:
I made the following function 
```
def _sort(x: list[i32]):
    k: i32 = len(x)
    i: i32
    j: i32
    for i in range(k):
        for j in range(i+1,k):
            if x[i]>x[j]:
                x[i],x[j]=x[j],x[i]
```
and passed the list by reference in the following manner:
```
_sort({x})
```

However, when I try to overload the sort function for f64, it throws an error. Any suggestions for alternative workarounds? 
@Smit-create